### PR TITLE
[bugfix] Group defaults to WORLD if None

### LINKED ISSUE #5122: Thre process  group doesn't default to all processes (world) but to None
## 🐛 Bug

If we correct the issue https://github.com/PyTorchLightning/pytorch-lightning/issues/5120#issue-766222158, and use the all_gather function of the LightningModule, we get another AttributeError: "NoneType object has no attribute allgather".

As stated in the documentation, the group parameter of the function:
https://github.com/PyTorchLightning/pytorch-lightning/blob/127454ade2b851dd267b7f0b4d973bdefd0329e5/pytorch_lightning/utilities/distributed.py#L192

defaults to all processes, but this is not the case. It defaults to None.

## Please reproduce using [the BoringModel and post here](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing)

https://colab.research.google.com/drive/1D7CnhxZziUoxTvqVqqsp_lJHHstQfz1r?usp=sharing

### Expected behavior

*The group parameter should be changed to group.WORLD if None.

### Environment

**Note**: `Bugs with code` are solved faster ! `Colab Notebook` should be made `public` !

* `IDE`: Please, use our python [bug_report_model.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py
) template.

* `Colab Notebook`: Please copy and paste the output from our [environment collection script](https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py) (or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```

 * CUDA:
	- GPU:
		- Tesla P100-PCIE-16GB
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.18.5
	- pyTorch_debug:     True
	- pyTorch_version:   1.7.0+cu101
	- pytorch-lightning: 1.1.0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020

### Additional context

Note that this issue is a logical contination of the issue https://github.com/PyTorchLightning/pytorch-lightning/issues/5120#issue-766222158 if it were corrected. In order to reproduce this issue, `all_gather` and `all_gather_ddp_if_available` were modified. Please see the comments in the colab for clarification.

### PR DESCRIPTION
**Pull request by 8greg8**
## What does this PR do?

Fixes #5122 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]
lgtm :]

**Review by SkafteNicki — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]
Saw one more bug down a bit.

File: pytorch_lightning/utilities/distributed.py [hunk 1]
```diff
@@ -203,10 +203,12 @@ def all_gather_ddp_if_available(
     Return:
         A tensor of shape (world_size, batch, ...)
     """
+    group = group if group is not None else torch.distributed.group.WORLD
+
```
Comment (rohitgr7) [inline pytorch_lightning/utilities/distributed.py:207]:
```suggestion

```

File: pytorch_lightning/utilities/distributed.py [hunk 2]
```diff
@@ -203,10 +203,11 @@ def all_gather_ddp_if_available(
     Return:
         A tensor of shape (world_size, batch, ...)
     """
+    group = group if group is not None else torch.distributed.group.WORLD
     if torch.distributed.is_available() and torch.distributed.is_initialized():
         if sync_grads:
             return AllGatherGrad.apply(tensor, group)
         else:
-            with torch.no_grad:
+            with torch.no_grad():
```
Comment (rohitgr7) [inline pytorch_lightning/utilities/distributed.py:211]:
```suggestion
            with torch.no_grad:
```

**Review by rohitgr7 — COMMENTED** [review]
oops my bad... there's another PR open for that. Putting the bug back

**Review by justusschock — APPROVED** [review]

**Review by ydcjeff — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/utilities/distributed.py [hunk 3]
```diff
@@ -203,6 +203,7 @@ def all_gather_ddp_if_available(
     Return:
         A tensor of shape (world_size, batch, ...)
     """
+    group = group if group is not None else torch.distributed.group.WORLD
```
Comment (SkafteNicki) [inline pytorch_lightning/utilities/distributed.py:206]: Should this be moved inside the `if` statement as `torch.distributed.group.WORLD` is not defined on windows?

**Comment by s-rog** [comment]
```
def all_gather_ddp_if_available(
    tensor: Union[torch.Tensor], group: Any = torch.distributed.group.WORLD, sync_grads: bool = False
) -> torch.Tensor:
```
This might be simpler?

**Comment by Borda** [comment]
> ```
> def all_gather_ddp_if_available(
>     tensor: Union[torch.Tensor], group: Any = torch.distributed.group.WORLD, sync_grads: bool = False
> ) -> torch.Tensor:
> ```
> 
> This might be simpler?

yes, just not sure about it some uses do already uses pass the None...

**Comment by s-rog** [comment]
I thought `None` produces an error as outlined in the linked issue, so this shouldn't break any existing working code?

**Pull request merged**